### ISSUE TITLE
Missing items after updating bookstack

### ISSUE THREAD

**Issue by uprightcarrion** [BookStackApp/BookStack#3660]
### Attempted Debugging

- [X] I have read the debugging page

### Searched GitHub Issues

- [X] I have searched GitHub for the issue.

### Describe the Scenario

After updating several things no longer show up on the webpage.  They are in the database and as far as I can tell the tables seem right.

For example here is the list of books in the table. But they don't show in my web page.  The user ID i'm using is 3 so I own them all
![image](https://user-images.githubusercontent.com/7152455/185471634-607b0e10-2117-414b-aa07-b5c183111196.png)
 Only 3 books show up
![image](https://user-images.githubusercontent.com/7152455/185472174-46f7722f-ae39-4f09-8620-ccca5d7f6ff1.png)

### Exact BookStack Version

BookStack v22.07.3 

### Hosting Environment

Ubuntu 20.04

**Comment by uprightcarrion** [comment]
The pages from the missing books are still visible under my user profile and i can view them.  When I view them and attempt to use the breadcrumb to access the book it says the book doesn't exist or I don't have permission to view it.  I can find a permission that could be causing it.  I own the books and I'm also setup with all the roles, admin editor, viewer, even public.

**Comment by ssddanbrown** [comment]
Hi @uprightcarrion,
That is strange.

Can you try running the command `php artisan bookstack:regenerate-permissions` from your BookStack install folder, and let me know if that resolves things.

**Comment by uprightcarrion** [comment]
That worked! Thank you!

What do you think happened?

**Comment by ssddanbrown** [comment]
Access rights to content in BookStack is essentially cached via another table (`joint_permissions`). It's this that really controls visibility, at least for scenarios where items are shown as a list. That fact that work indicates this was out of sync with the actual permissions state. 

Why that was, I'm not totally sure. There have been some heavy changes to the `joint_permissions` table, in the v22.04 and v22.07, but none of the changes should have affected existing view permissions unless something went astray or was interrupted.

Just to help me keep an eye on any similar reports and consider what might of occurred, could you confirm the following?:

- Do you know what version you upgraded from?
- How is your Ubuntu system running? Is it a VM of some kind or using a specific VPS provider?

**Comment by uprightcarrion** [comment]
I upgraded from 22.03 

It's running on a Digital Ocean VPS, Ubuntu 20.04

I installed it using the composer method

This is the second time I've upgraded it, the first time no problems.

**Comment by ssddanbrown** [comment]
Thanks for confirming those details.
Since this is solved I'll close this off but I'll keep this in mind to see if this becomes common at all, and those details will help see any patterns.

**Comment by tmrhymer** [comment]
I encountered this same issue today. We ended up rolling back using a VM snapshot because I didn't find this post quick enough. Here is what our environment looked like

    Do you know what version you upgraded from?
v22.06.2 upgrading to v22.07.3

    How is your Ubuntu system running? Is it a VM of some kind or using a specific VPS provider?
We are on Centos 7 running on VMware.

We're going to attempt the upgrade again soon and I'll keep that command in mind. Should we run this before attempting?
`php artisan bookstack:regenerate-permissions`

**Comment by ssddanbrown** [comment]
@jimmyc802 Probably better to run the command after updating.

**Comment by ssddanbrown** [comment]
Documenting some findings here based upon other user reports, for possible future reference.

This scenario could have occurred if an instance was updated to BookStack >= v22.07 but not migrated through the full pending migration set. Any permissions changes within the instance would then be recorded assuming the new structure, leaving an empty `action` in `joint_permissions`. Upon next migration run, these entries would then be deleted causing content to not show until permissions were regenerated.

This explains the variance in versions where this is experienced, and the variation of content that's hidden post upgrade.

Note, there may be other possible scenarios, but the above is the only one I've reproduced so far.
